colossalai/utils/profiler/common.py

```diff
@@ -0,0 +1,75 @@
+from enum import Enum
+from .comm_profiler import CommProfiler
+from typing import List
+from torch.utils.tensorboard import SummaryWriter

```

Comment:
should catch import error here as this requires tensorboard installation

Reply:
Pytorch already helps raise the import error

Reply:
This line makes TensorBoard a must.

Reply:
Do you mean even there is no tensorboard installed we still make profiler work?

Reply:
Yes

Reply:
In fact, a better workaround of this is to import tensorboard only in `to_tensorboard`. Type hinting can be given as a string e.g. 

```python
writer: "torch.utils.tensorboard.SummaryWriter"
```